src/driver-marsgaming/marsgaming-protocol.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+#include "marsgaming-buttons.h"
+#include "marsgaming-leds.h"
+
+enum marsgaming_report_type {
+	MARSGAMING_MM4_REPORT_TYPE_UNKNOWN_1 = 0x01,
+	MARSGAMING_MM4_REPORT_TYPE_WRITE = 0x02,
+	MARSGAMING_MM4_REPORT_TYPE_READ = 0x03,
+	MARSGAMING_MM4_REPORT_TYPE_UNKNOWN_4 = 0x04,
+	MARSGAMING_MM4_REPORT_TYPE_UNKNOWN_6 = 0x06
+};
+
+struct marsgaming_report_resolution_info {
+	bool enabled;
+	uint16_t x_res;
+	uint16_t y_res;
+	// 4 lowest bits, each one corresponds to one led
+	// Resolution 0 -> b0000
+	// Resolution 1 -> b0001
+	// Resolution 2 -> b0011
+	// etc
+	uint8_t led_bitset;
+	uint8_t zeros_3[2];
+} __attribute__((packed));
+
+struct marsgaming_report_resolutions {
+	uint8_t usb_report_id;
+	uint8_t report_type; // marsgaming_report_type
+	uint8_t unknown_2; // 0x4f
+	uint8_t profile_id;
+	uint8_t unknown_4; // 0x2a
+	uint8_t unknown_5; // 0x00
+	uint8_t unknown_6; // 0x00 from device | 0xfa from host
+	uint8_t unknown_7; // 0x00 from device | 0xfa from host
+	uint8_t count_resolutions;
+	uint8_t current_resolution;
+	union {
+		struct marsgaming_report_resolution_info resolutions[6];
+		uint8_t padding[54];
+	};
```

Comment:
```c
	struct marsgaming_report_resolution_info resolutions[6];
	uint8_t padding[6];
```
Here too, you can put the padding after the data.

https://github.com/libratbag/libratbag/pull/1364/files/14a2a85dd0f40a18afac41761aeff69339608552#r1090238728